modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -252,23 +283,27 @@ private void initConfigurationResolver() {
 
     if (isConfigurationSpecified()) {
       // the config is dynamic
-      configurationResolver = event -> {
-        ConfigurationInstance instance = configurationProvider.get().get(event);
-        if (instance == null) {
-          throw new IllegalModelDefinitionException(format(
-                                                           "Root component '%s' contains a reference to config '%s' but it doesn't exists",
-                                                           getLocation().getRootContainerName(),
-                                                           configurationProvider));
-        }
-
-        return of(instance);
-      };
+      // the optional should be always present at this point because we already checked dependsOnEvent
+      configurationResolver = event -> resolveConfigFromProvider(getConfigurationProvider().get(), event);
     } else {
       // obtain implicit instance
       configurationResolver = event -> extensionManager.getConfiguration(extensionModel, componentModel, event);
     }
   }
 
+  private Optional<ConfigurationInstance> resolveConfigFromProvider(ConfigurationProvider configurationProvider,
```

Comment:
this method name is not clear. It's not just resolving config from provider, there's an underlying assumption that this method is only to be used for dynamic configs.

Reply:
Actually it is currently used both for static and dynamic configs, because when the expression is resolved and we have the `ConfigurationProvider`, it can be any of those.
I'm not sure if it would be helpful to create different versions for static and dynamic configs only to avoid the null check for static ones.
The only cases in which we don't go here is when there is no expression in the config-ref **and** the ConfigurationProvider is static, or when using implicit configs.

Reply:
I added some comments to make it clearer that the method may be used for both static and dynamic config providers. I still think the name of the method is representative of what it does. Maybe we can change it to `getConfigFromProvider` if that sounds better. Basically that's all it does, but with some added error handling (which I don't even know if it is actually needed, but I kept just in case).